nuxeo-page-provider-display-behavior.html

```diff
@@ -514,15 +514,37 @@
       _fetchRange(firstIndex, lastIndex, clear) {
         if (this._hasPageProvider()) {
 
+          if (firstIndex === 0) {
+            lastIndex = this._pageSize - 1;
```

Comment:
I understand defining a value for `lastIndex` if it's not defined, but why do we need to override it here?

Reply:
Because when `firstIndex` is equals `0` it is likely the search params have changed and we are displaying the very first results again.

In that case, I've noticed locally sometimes that the `lastIndex` isn't as high as needed to fully populate the visible range and we end up with missing visible items and an extra requests to fetch them. Doing so addresses this "glitch"

Reply:
If I understand correctly, `_fetchRange` is called twice, on `fetch` and `_fetchRange`. Given the first already uses those values explicitly (i.e. `this._fetchRange(0, this._pageSize - 1, true);`, perhaps we should use this logic instead on `_scrollChanged`? If not, I think we should at least document why this if is there, to prevent happening to it the same thing that happened to the following if. 😝